Sources/Commands/PackageCommands/AddDependency.swift

```diff
@@ -52,31 +53,56 @@ extension SwiftPackageCommand {
 
         func run(_ swiftCommandState: SwiftCommandState) throws {
             let workspace = try swiftCommandState.getActiveWorkspace()
-
             guard let packagePath = try swiftCommandState.getWorkspaceRoot().packages.first else {
                 throw StringError("unknown package")
             }
 
-            // Load the manifest file
-            let fileSystem = workspace.fileSystem
-            let manifestPath = packagePath.appending("Package.swift")
-            let manifestContents: ByteString
-            do {
-                manifestContents = try fileSystem.readFileContents(manifestPath)
-            } catch {
-                throw StringError("cannot find package manifest in \(manifestPath)")
+            if let url = SourceControlURL(absoluteString: dependency) {
```

Comment:
We also have a `CanonicalPackageURL` that @francescomikulis is changing a little in https://github.com/swiftlang/swift-package-manager/pull/7741. What if instead of the new `SourceControlURL` here, we use `CanonicalPackageURL` and then check if the scheme is either `file` or `nil`?